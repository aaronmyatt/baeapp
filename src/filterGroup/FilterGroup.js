import React, { Component } from 'react'

class FilterGroup extends Component {
    constructor(props){
        super(props)
        this.state = {activeGroup:[]}

        this.locations = this.props.locations
        this.activeGroup = this.props.activeGroup
        this.service = this.props.service

        this.groupButtonClickHandler = this.groupButtonClickHandler.bind(this)
        this.uniqueGroupValueButtonClickHandler = this.uniqueGroupValueButtonClickHandler.bind(this)
    }

    groupButtonClickHandler(e) {
      this.setState({activeGroup:[e.target.innerText]})
    }

    uniqueGroupValueButtonClickHandler(e) {
        this.setState({activeGroup:[this.state.activeGroup[0], e.target.innerText]})
        let activeLocations = this.service.filterByGroup(this.locations, this.state.activeGroup)
        this.props.setActiveLocations(activeLocations)
    }

    renderUniqueGroupValues() {
        let groups = this.service.getUniqueGroupValues(this.locations, this.state.activeGroup[0])
          return groups.map((n, index)=>{
              return < FilterButton key={index} name={n} handler={this.uniqueGroupValueButtonClickHandler}/>
          })
    }

    renderGroupButtons() {
        let groups = this.service.getGroups(
            this.locations,
            ["id", "address", "lng", "lat", "postalCode"])
        return groups.map((n, index)=>{
            return < FilterButton key={n} name={n} handler={this.groupButtonClickHandler} />
        })
    }

    render() {
      return (
          <div>
              <div>
                  <ul>{this.renderGroupButtons()}</ul>
              </div>
              <div>
                  <ul>{this.renderUniqueGroupValues()}</ul>
              </div>
          </div>
      );
    }
}

class FilterButton extends Component {
    constructor(props){
        super(props)
        this.state = {}
        this.handler = this.props.handler.bind(this)
    }

  render() {
      let name = this.props.name
    return (
        <li>
            <button onClick={this.handler} >{name}</button>
        </li>
    );
  }
}

export default FilterGroup
