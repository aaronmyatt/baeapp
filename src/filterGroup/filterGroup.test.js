import React from 'react'
import renderer from 'react-test-renderer'
import locService from '../locationsService/LocationsService'
import locations from '../data/MockLocations'
import FilterGroup from './FilterGroup'


describe("filter group component", () => {

    test('renders static content', () => {
      const props = {locations:locations, service:locService}
      const output = renderer.create(< FilterGroup {...props} />);
      expect( output.toJSON() ).toMatchSnapshot()
    })

})
