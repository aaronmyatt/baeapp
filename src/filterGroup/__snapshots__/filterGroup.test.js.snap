exports[`filter group component renders static content 1`] = `
<div>
  <div>
    <ul>
      <li>
        <button
          onClick={[Function]}>
          city
        </button>
      </li>
      <li>
        <button
          onClick={[Function]}>
          country
        </button>
      </li>
    </ul>
  </div>
  <div>
    <ul>
      <li>
        <button
          onClick={[Function]} />
      </li>
    </ul>
  </div>
</div>
`;
